feat(MMPD-546): update portfolio icon

GuiBibeau opened the pull request:
<!--
Thanks for your contribution!

Please ensure that any applicable requirements below are satisfied before submitting this pull request. This will help ensure a quick and efficient review cycle.
-->

**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

Due to a redesign, we are changing the link to portfolio from a simple icon to instead be a secondary button. 

**Screenshots/Recordings**

<img width="385" alt="Screenshot 2023-07-06 at 12 52 45 PM" src="https://github.com/MetaMask/metamask-mobile/assets/22625602/1e2af7ff-80b3-4d1c-bb5b-11332a33d1fd">

<img width="375" alt="Screenshot 2023-07-12 at 4 47 40 PM" src="https://github.com/MetaMask/metamask-mobile/assets/22625602/cf12a69e-87cc-45e3-8f13-654b2e0154b1">


How to test:

1. Go to the main view
2. click the portfolio button and make sure the portfolio link works like before

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

GuiBibeau committed:
feat(MMPD-546): update portfolio icon

sethkfman commented on app/components/UI/Tokens/index.tsx:
```diff
@@ -459,17 +459,15 @@ const Tokens: React.FC<TokensI> = ({ tokens }) => {
+          style={styles.buyButton}
           onPress={onOpenPortfolio}
-          style={styles.portfolioLink}
-          {...generateTestId(Platform, PORTFOLIO_BUTTON)}
```
I would make sure this ID still exists in the new component

GuiBibeau replied:
Good call, I added it to the new component too.

sethkfman commented on app/components/UI/Tokens/index.tsx:
```diff
@@ -393,7 +393,7 @@ const Tokens: React.FC<TokensI> = ({ tokens }) => {
           style={styles.buyButton}
           onPress={goToBuy}
-          label={strings('wallet.buy_asset', { asset: mainToken.symbol })}
+          label={strings('wallet.portfolio_button')}
```
Is the buy button being replaced?

GuiBibeau replied:
Oh sorry that's a mistake on my part that was supposed to go to another button. Updating.

sethkfman commented on app/components/UI/Tokens/index.tsx:
```diff
@@ -459,17 +459,15 @@ const Tokens: React.FC<TokensI> = ({ tokens }) => {
-          style={styles.portfolioLink}
-          {...generateTestId(Platform, PORTFOLIO_BUTTON)}
-        >
-          <Icon
```
I would remove these components from the import. Run `yarn lint` and confirm you don't have any errors.

GuiBibeau replied:
Good point, ran it and remove imports

sethkfman commented:
@yanrong-chen Can you take a look at the UI above to make sure it is aligned with platform design?

GuiBibeau committed:
feat(MMPD-546): lint, update to use translations, keep test id

GuiBibeau committed:
Merge branch 'main' into update-portfolio-icon

GuiBibeau commented on app/components/UI/Tokens/index.tsx:
```diff
@@ -457,17 +453,16 @@ const Tokens: React.FC<TokensI> = ({ tokens }) => {
-            size={IconSize.Md}
-          />
-        </TouchableOpacity>
+          endIconName={IconName.Login}
```
This is the icon that might change.

yanrong-chen commented:
> @yanrong-chen Can you take a look at the UI above to make sure it is aligned with platform design?

Hi Seth, we are using a different icon in this portfolio entry button
![portfolio entry(1)](https://github.com/MetaMask/metamask-mobile/assets/59982439/252c46d8-e275-459e-bf0a-0204e4523699)

jclancy93 commented:
LGTM from the portfolio side

GuiBibeau commented:
@yanrong-chen @sethkfman I'm currently waiting to get Figma access to have the right account to put here. I left a comment on this in the PR description.
https://github.com/MetaMask/metamask-mobile/pull/6751#issuecomment-1623975440

yanrong-chen commented:
> @yanrong-chen @sethkfman I'm currently waiting to get Figma access to have the right account to put here. I left a comment on this in the PR description. [#6751 (comment)](https://github.com/MetaMask/metamask-mobile/pull/6751#issuecomment-1623975440)

I moved this design to the following link, pls let me know if you still can't access the page 
https://www.figma.com/file/fFuWPCTvJ9ZYHIxFyRq0HH/portfolio-entry?type=design&node-id=0-1&mode=design&t=ayqoKxzA7R2vFqu3-0

GuiBibeau committed:
Update index.tsx

change icon

GuiBibeau commented:
> > @yanrong-chen @sethkfman I'm currently waiting to get Figma access to have the right account to put here. I left a comment on this in the PR description. [#6751 (comment)](https://github.com/MetaMask/metamask-mobile/pull/6751#issuecomment-1623975440)
> 
> I moved this design to the following link, pls let me know if you still can't access the page https://www.figma.com/file/fFuWPCTvJ9ZYHIxFyRq0HH/portfolio-entry?type=design&node-id=0-1&mode=design&t=ayqoKxzA7R2vFqu3-0

Thanks that gave me the name! I updated the code and screenshot

GuiBibeau committed:
fix snapshots

GuiBibeau committed:
Merge branch 'main' into update-portfolio-icon

yanrong-chen commented:
Hi @GuiBibeau - The new screenshot looks great to me! Thank you for the fix.

GuiBibeau committed:
Merge branch 'main' into update-portfolio-icon

MarioAslau reviewed (approved):
lgtm !

GuiBibeau committed:
Merge branch 'main' into update-portfolio-icon

GuiBibeau committed:
Merge branch 'main' into update-portfolio-icon
